crates/goose/src/developer.rs

```diff
@@ -133,33 +115,34 @@ impl DeveloperSystem {
 
     // Implement bash tool functionality
     async fn bash(&self, params: Value) -> AgentResult<Value> {
+        let working_dir = params
+            .get("working_dir")
+            .and_then(|v| v.as_str())
+            .unwrap_or(".");
         let command = params.get("command").and_then(|v| v.as_str());
         let source_path = params.get("source_path").and_then(|v| v.as_str());
-        let working_dir = params.get("working_dir").and_then(|v| v.as_str());
 
-        if command.is_none() && source_path.is_none() && working_dir.is_none() {
+        if command.is_none() && source_path.is_none() {
             return Err(AgentError::InvalidParameters(
-                "At least one parameter must be provided".into(),
+                "At least one of 'command' or 'source_path' must be provided".into(),
             ));
         }
 
         let mut outputs = Vec::new();
 
         // Change working directory
```

Comment:
not for this pr, but let's try an experiment sometime where CWD and source are scoped to just one command and do not change globally. we've never tried that variation, i think it might be more consistent

Reply:
yeah i was thinking we could rewrite the command to be `source {source_path}; {command}` and make it a single command